packages/client-sdk-nodejs/src/config/retry/fixed-timeout-retry-strategy.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DeterminewhenToRetryRequestProps,
+  RetryStrategy,
+} from './retry-strategy';
+import {EligibilityStrategy} from './eligibility-strategy';
+import {
+  MomentoLoggerFactory,
+  MomentoLogger,
+  DefaultMomentoLoggerFactory,
+} from '../..';
+import {DefaultStorageEligibilityStrategy} from './storage-default-eligibility-strategy';
+
+export interface FixedTimeoutRetryStrategyProps {
+  loggerFactory: MomentoLoggerFactory;
+  eligibilityStrategy?: EligibilityStrategy;
+
+  // Retry request after a fixed time interval (defaults to 100ms)
+  retryDelayIntervalMillis?: number;
+
+  // Number of milliseconds the client is willing to wait for response data to be received before retrying (defaults to 1000ms). After the overarching GRPC config deadlineMillis has been reached, the client will terminate the RPC with a Cancelled error.
+  responseDataReceivedTimeoutMillis?: number;
+}
+
+export class FixedTimeoutRetryStrategy implements RetryStrategy {
+  private readonly logger: MomentoLogger;
+  private readonly eligibilityStrategy: EligibilityStrategy;
+  private readonly retryDelayIntervalMillis: number;
+  readonly responseDataReceivedTimeoutMillis: number;
+
+  constructor(props: FixedTimeoutRetryStrategyProps) {
+    this.logger = props.loggerFactory.getLogger(this);
+    this.eligibilityStrategy =
+      props.eligibilityStrategy ??
+      new DefaultStorageEligibilityStrategy(props.loggerFactory);
+    this.retryDelayIntervalMillis = props.retryDelayIntervalMillis ?? 100;
+    this.responseDataReceivedTimeoutMillis =
+      props.responseDataReceivedTimeoutMillis ?? 1000;
+  }
+
+  determineWhenToRetryRequest(
+    props: DeterminewhenToRetryRequestProps
+  ): number | null {
+    this.logger.debug(
+      `Determining whether request is eligible for retry; status code: ${props.grpcStatus.code}, request type: ${props.grpcRequest.path}, attemptNumber: ${props.attemptNumber}`
+    );
+    if (!this.eligibilityStrategy.isEligibleForRetry(props)) {
+      // null means do not retry
+      return null;
+    }
+
+    this.logger.debug(
+      `Request is eligible for retry (attempt ${props.attemptNumber}), retrying soon.`
+    );
+    // retry after a fixed time interval has passed (+/- some jitter)
+    return addJitter(this.retryDelayIntervalMillis);
+  }
+}
+
+export class FixedTimeoutRetryStrategyFactory {
+  static getRetryStrategy(
+    props?: FixedTimeoutRetryStrategyProps
+  ): FixedTimeoutRetryStrategy {
+    return new FixedTimeoutRetryStrategy({
+      loggerFactory: props?.loggerFactory ?? new DefaultMomentoLoggerFactory(),
```

Comment:
I don't think we can do this (default value for logger factory) - it can result in them ending up with two different logging destinations. Hopefully in the pre-built configurations we are constructing the strategy using the single logger factory that gets passed in there. And I think we just have to lean into the fact that if someone wants to customize the strategy they will have to pass in a logger factory.

and then, if we remove that, this factory function isn't really any more succinct than the main constructor so I think we can ditch it.

I think with the state you've gotten this stuff to now, the constructor is pretty approachable/reasonable if someone wants to customize the arguments, so we are probably good there. But I will pull down the code and get my hands dirty with the API/UX for a few minutes to confirm.